examples/query/react/basic/src/store.ts

```diff
@@ -1,11 +1,28 @@
-import { configureStore } from '@reduxjs/toolkit'
+import {
+  combineReducers,
+  configureStore,
+  getDefaultMiddleware,
+} from '@reduxjs/toolkit'
+import type { PreloadedState } from '@reduxjs/toolkit'
 import { pokemonApi } from './services/pokemon'
 
-export const store = configureStore({
-  reducer: {
-    [pokemonApi.reducerPath]: pokemonApi.reducer,
-  },
-  // adding the api middleware enables caching, invalidation, polling and other features of `rtk-query`
-  middleware: (getDefaultMiddleware) =>
-    getDefaultMiddleware().concat(pokemonApi.middleware),
+const rootReducer = combineReducers({
+  [pokemonApi.reducerPath]: pokemonApi.reducer,
 })
+
+export const setupStore = (initialState?: PreloadedState<RootState>) => {
+  // Adding pokemonApi's middleware to default middleware for RTK-Query features
+  const middleWare = getDefaultMiddleware<RootState>().concat(
+    pokemonApi.middleware
+  )

```

Comment:
I would just use the callback notation within `configureStore` for `getDefaultMiddleware` rather than the exported version. You save an import, and don't need to give it the `RootState` type explicitly

Reply:
:100: we are actually planning on deprecating the `getDefaultMiddleware` import in favor of the callback notation.

Reply:
good to know!